cmd/key_rotate.go

```diff
@@ -0,0 +1,162 @@
+package cmd
+
+import (
+	"errors"
+	"fmt"
+	"io"
+
+	"code.vegaprotocol.io/vegawallet/cmd/cli"
+	"code.vegaprotocol.io/vegawallet/cmd/flags"
+	"code.vegaprotocol.io/vegawallet/cmd/printer"
+	"code.vegaprotocol.io/vegawallet/wallet"
+	"code.vegaprotocol.io/vegawallet/wallets"
+	"github.com/spf13/cobra"
+)
+
+var (
+	rotateKeyLong = cli.LongDesc(`
+		Get a signed key rotation transaction as a Base64 encoded string by choosing a new public key to rotate to and target block height.
+
+		Later the transaction is applied to Vega protocol through the wallet's "send tx" command.
+	`)
+
+	rotateKeyExample = cli.Examples(`
+		Given that those variables exists:
+			- WALLET - Vega wallet to be used.
+			- PUBLIC_KEY - A newly generated public key. Should be generate by wallet's "generate" command.
+			- TARGET_HEIGHT - Height of block where the new public key change will take effect
+			- TX_HEIGHT - It should be close to the current block height when the transaction is applied, with a threshold of ~ - 150 blocks.
```

Comment:
I would argue that this is just duplication of the description already against each option. Do we need it in two places?
```
Flags:
  -h, --help                     help for rotate
  -p, --passphrase-file string   Path to the file containing the wallet's passphrase
  -t, --target-height uint32     Block height from which the new public key will be used
  -x, --tx-height uint32         Block height of which the transaction will be applied at
  -w, --wallet string            Wallet holding the master key and new public key
```

Reply:
Hum that's true indeed. I think you are right.

Reply:
Also I'm still against the use of the word "new". It being a a newly generated public key isn't a requirement. Someone could generated 1000 keys upfront, save the wallets offline somewhere and rotate through them over 2 years. Also someone who isn't a native speaker could become confused and think that it is `key rotate` that is generating the new public key. 

What is really happen is that they are rotating to the _next_ key in the chain.

Reply:
@wwestgarth Yeah, well I think it depends. Because it is a new key in the context of the transaction. That being said the phrase `A newly generated public key` is not exactly right.. I agree with that.

Reply:
@ValentinTrinque the reason why we need transaction height here is because the machine where this command will be run should be disconnected from internet so we can't fetch the block height. Not sure whether this needs to be a pattern for all other commanda?

Reply:
@karlem That was not for TX_HEIGHT specifically. But, I get your point ;)